src/prob/pf.jl

```diff
@@ -198,3 +198,803 @@ function build_mc_pf(pm::AbstractUBFModels)
+                c_nl_func = function(v_bt)
+                    vd = Md*v_bt
+                    pd = load["pd"].*(abs.(vd)./load["vnom_kv"]).^load["alpha"]
+                    qd = load["qd"].*(abs.(vd)./load["vnom_kv"]).^load["beta"]
+                    sd = pd+im*qd
+                    cd = conj.(sd./vd)
+                    cd_bus = Md'*cd
+                    return -(cd_bus .- y_prim*v_bt)
+                end
+                c_tots_func = function(v_bt)
+                    return y_prim * v_bt .+ c_nl_func(v_bt)
+                end
+            end
+        end
+    end
+
+    return bts, 0, y_prim, c_nl_func, c_tots_func
+end
+
+
+"""
+    _cpf_generator_interface(
+      gen::Dict,
+      v_start::Dict
+    )
+
+Generator component interface outputs generator primitive Y matrix.
+"""
+function _cpf_generator_interface(gen::Dict{String,<:Any}, v_start::Dict{<:Any,<:Any}, explicit_neutral::Bool)
+    bts = [(gen["gen_bus"], t) for t in gen["connections"]]
+    wires = length(bts)
+    v0_bt = [v_start[bt] for bt in bts]
+
+    conf = gen["configuration"]
+
+    if conf==WYE
+        if explicit_neutral
+            vd0 = v0_bt[1:end-1] .- v0_bt[end]
+        else
+            vd0 = v0_bt
+        end
+        sg0 = gen["pg"]+im*gen["qg"]
+        sd0 = -sg0
+        c0 = conj.(sd0./vd0)
+        y0 = c0./vd0
+        if explicit_neutral
+            y_prim = [diagm(y0) -y0; -transpose(y0) sum(y0)]
+        else
+            y_prim = diagm(y0)
+        end
+        
+        c_nl_func = function(v_bt)
+            sg = gen["pg"]+im*gen["qg"]
+            sd = -sg
+            if explicit_neutral
+                vd = v_bt[1:end-1].-v_bt[end]
+                cd = conj.(sd./vd)
+                cd_bus = [cd..., -sum(cd)]
+                return -(cd_bus .- y_prim*v_bt)
+            else
+                vd = v_bt
+                cd = conj.(sd./vd)
+                cd_bus = cd
+                return -(cd_bus .- y_prim*v_bt)                
+            end
+        end
+        c_tots_func = function(v_bt)
+            return y_prim * v_bt .+ c_nl_func(v_bt)
+        end
+
+    elseif conf==DELTA
+        if length(v0_bt) == 3
+            Md = [1 -1 0; 0 1 -1; -1 0 1]
+        elseif length(v0_bt) == 2
+            Md = [1 -1]
+        end
+        vd0 = Md*v0_bt
+        sg0 = gen["pg"]+im*gen["qg"]
+        sd0 = -sg0
+        c0 = conj.(sd0./vd0)
+        y0 = c0./vd0
+        y_prim = Md'*diagm(0=>y0)*Md
+        c_nl_func = function(v_bt)
+            sg = gen["pg"]+im*gen["qg"]
+            sd = -sg
+            vd = Md*v_bt
+            cd = conj.(sd./vd)
+            cd_bus = Md'*cd
+            return -(cd_bus .- y_prim*v_bt)
+        end
+        c_tots_func = function(v_bt)
+            return y_prim * v_bt .+ c_nl_func(v_bt)
+        end
+    end
+
+    return bts, 0, y_prim, c_nl_func, c_tots_func
+end
+
+
+"""
+    _cpf_switch_interface(
+      switch::Dict,
+      v_start::Dict
+    )
```

Comment:
missing epxplicit_neutral in docstring (probably do a find all, I may have missed some)

Reply:
updated all docstrings